docs/architecture/adr-074-timeout-params.md

```diff
@@ -0,0 +1,228 @@
+	no quorum before entering the next round.
+* `TimeoutPrecommitDelta` 
+	* How much the `TimeoutPrecommit` increases with each round.
+* `TimeoutCommit` 
+	* How long the consensus algorithm waits after committing a block but before starting the new height.
+	* This gives a validator a chance to receive slow precommits.
+* `SkipTimeoutCommit`
+	* Make progress as soon as the node has 100% of the precommits.
+
+
+### Overview of Proposed Change
+
+This ADR proposes consolidating the timeout parameters and migrating them from the
+node-local `config.toml` file into the network-global consensus parameters.
+The consensus parameters are the same across all validators on the network.
+
+A sane default will be provided by Tendermint for each of these parameters and
+networks will be able to update the parameters as they see fit. Local updates
+to these parameters will no longer be possible; instead, the application will control
+updating the parameters. Applications using the Cosmos SDK will be automatically be
+able to change the values of these consensus parameters [via a governance proposal][cosmos-sdk-consensus-params].
+This ADR also proposes consolidating the 8 timeout parameters into 6. These
+6 parameters would be as follows:
+
+* `TimeoutPropose`
+	* Same as current `TimeoutPropose`.
+* `TimeoutProposeDelta`
+	* Same as current `TimeoutProposeDelta`.
+* `TimeoutVote`
+	* Parameter that governs how long validators wait for votes in both the prevote
+	 and precommit phase of the consensus algorithm. This parameter will subsume
+	 the current `TimeoutPrevote` and `TimeoutPrecommit` parameters.
+* `TimeoutVoteDelta`
+	* Parameter that governs how much the `TimeoutVote` will grow each successive round.
+	 This parameter will subsume the current `TimeoutPrevoteDelta` and `TimeoutPrecommitDelta`
+	 parameters.
+* `TimeoutCommit`
+	* Same as current `TimeoutCommit`.
+* `SkipTimeoutCommit`
+	* Same as current `SkipTimeoutCommit`.
+
+This change is low risk. While these values may be configured, many running chains
+do not change these parameters from their default values. For example, initializing
+a node on Osmosis, Terra, and the Cosmos Hub using the their `init` command produces
+a `config.toml` with Tendermint's default values for these parameters.
+
+### Why this parameter consolidation?
+
+Reducing the number of parameters is good for UX. Fewer superfluous parameters makes
+running and operating a Tendermint network less confusing. 
+
+The Prevote and Precommit messages are both similar sizes and are also similar in
+their function in the consensus algorithm so there is no strong need for them to
+be configured separately.
+
+The `TimeoutPropose` parameter governs how long Tendermint will wait for the proposed
+block to be gossiped. Blocks are much larger than votes and therefore tend to be
+gossiped much more slowly. It therefore makes sense to keep `TimeoutPropose` and
+the `TimeoutProposeDelta` as parameters separate from the vote timeouts.
```

Comment:
Yeah, it effectively allows the algorithm to 'search' for the correct timeouts so that it can adapt to changes in network conditions/topology and block size.

Reply:
Not that it matters for this ADR, but how does it adjust back down? Or does it reset to the base on every new block, discarding adjustments from the rounds of the previous height?

Reply:
Resets back down to the base at each height and discards the adjustments from the previous height.
There shouldn't be any issue with changing the logic to stay at the higher value for the next round or 'search' for a large enough time in a different way.

Reply:
And if it's consistently too short then now they the network can agree to adjust it to a longer time